File: Report_1/Step_1/Step_1.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <conio.h>

#define FMROW 500
#define FMCOL 250
#define SMROW 250
#define SMCOL 500

void main()
{
    FILE *f;
    int i, j, k;
    int fm[FMROW][FMCOL], sm[SMROW][SMCOL], res[FMROW][SMCOL];
    clock_t start, finish;
    double duration;

    f = fopen("Data01.txt", "r");
    printf("Reading Data01.txt\n");

    for(i = 0; i < FMROW; i++)
    {
        for(j = 0; j < FMCOL; j++)
        {
            fscanf(f, "%d", &fm[i][j]);
        }
    }

    fclose(f);

    f = fopen("Data02.txt", "r");
    printf("Reading Data02.txt\n");

    for(i = 0; i < SMROW; i++)
    {
        for(j = 0; j < SMCOL; j++)
        {
            fscanf(f, "%d", &sm[i][j]);
        }
    }

    fclose(f);

    printf("Calculating... loop k->j->i\n");
    start = clock();

    for(k = 0; k < SMROW; k++)
    {
        for(j = 0; j < SMCOL; j++)
        {
            for(i = 0; i < FMROW; i++)
            {
                res[i][j] += (fm[i][k] * sm[k][j]);
            }
        }
    }
    finish = clock();

    duration = (double)(finish - start) / CLOCKS_PER_SEC;

    printf("\n\n");
    printf("*** 총 소요시간 : %.3lf 초\n", duration);
    system("pause");
}

File: Report_1/Step_1/Step_1.java
import java.io.File;
import java.io.FileReader;
import java.io.IOException;

public class Step_1 {
	static int FMROW = 1000, FMCOL = 1000;
	static int SMROW = 1000, SMCOL = 1000;

   public static void main(String[] args) {
       int[][] fm = new int[FMROW][FMCOL];
       int[][] sm = new int[SMROW][SMCOL];
       int[][] result;

       try {

           File file = new File("./Data01.txt");
           FileReader filereader = new FileReader(file);

           for (int i = 0; i < FMCOL; i++) {
               for (int j = 0; j < FMROW; j++)
            	   fm[i][j] = filereader.read();
           }

           filereader.close();

           file = new File("./Data02.txt");
           filereader = new FileReader(file);

           for (int i = 0; i < SMCOL; i++) {
               for (int j = 0; j < SMROW; j++)
            	  sm[i][j] = filereader.read();
           }

           filereader.close();

           long startTime = System.currentTimeMillis();

           result = multiply(fm, sm);

           long endTime = System.currentTimeMillis();

           long lTime = endTime - startTime;
           System.out.println("TIME : " + lTime / 1000.0 + "(ms)");

       } catch (IOException e) {

    	   System.out.println(e);

       }
   }

   public static int[][] multiply(int[][] a, int[][] b) {
       int[][] c = new int[FMROW][SMCOL];

       for (int i = 0; i < FMROW; i++) {
           for (int j = 0; j < SMCOL; j++) {
               for (int k = 0; k < FMCOL; k++) {
                   c[i][j] = c[i][j] + a[i][k] * b[k][j];
               }
           }
       }

       return c;
   }
}
